test/pg-cdc/pg-cdc.td

```diff
@@ -0,0 +1,392 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+#
+# Prime the Postgresql database
+#
+
+$ postgres-execute connection=postgres://postgres:postgres@postgres
+ALTER USER postgres WITH replication;
+DROP PUBLICATION IF EXISTS mz_source;
+DROP SCHEMA IF EXISTS public CASCADE;
+CREATE SCHEMA public;
+
+CREATE TABLE pk_table (pk INTEGER PRIMARY KEY, f2 TEXT);
+INSERT INTO pk_table VALUES (1, 'one');
+ALTER TABLE pk_table REPLICA IDENTITY FULL;
+INSERT INTO pk_table VALUES (2, 'two');
+
+CREATE PUBLICATION mz_source FOR ALL TABLES;
+INSERT INTO pk_table VALUES (3, 'three');
+
+CREATE TABLE nonpk_table (f1 INTEGER, f2 INTEGER);
+INSERT INTO nonpk_table VALUES (1, 1), (1, 1);
+ALTER TABLE nonpk_table REPLICA IDENTITY FULL;
+INSERT INTO nonpk_table VALUES (2, 2), (2, 2);
+
+CREATE TABLE types_table (date_col DATE, time_col TIME, timestamp_col TIMESTAMP, uuid_col UUID, double_col DOUBLE PRECISION);
+INSERT INTO types_table VALUES ('2011-11-11', '11:11:11', '2011-11-11 11:11:11', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11', 1234.56768);
+ALTER TABLE types_table REPLICA IDENTITY FULL;
+
+CREATE TABLE large_text (f1 TEXT);
+INSERT INTO large_text VALUES (REPEAT('x', 16 * 1024 * 1024));
+ALTER TABLE large_text REPLICA IDENTITY FULL;
+
+CREATE TABLE trailing_space_pk (f1 TEXT PRIMARY KEY);
+INSERT INTO trailing_space_pk VALUES ('abc   ');
+ALTER TABLE trailing_space_pk REPLICA IDENTITY FULL;
+
+CREATE TABLE trailing_space_nopk (f1 TEXT);
+INSERT INTO trailing_space_nopk VALUES ('abc   ');
+ALTER TABLE trailing_space_nopk REPLICA IDENTITY FULL;
+
+CREATE TABLE multipart_pk(f1 INTEGER, f2 TEXT, f3 TEXT, PRIMARY KEY (f1, f2));
+INSERT INTO multipart_pk VALUES (1, 'abc', 'xyz');
+ALTER TABLE multipart_pk REPLICA IDENTITY FULL;
+
+CREATE TABLE nulls_table (f1 TEXT, f2 INTEGER);
+INSERT INTO nulls_table VALUES (NULL, NULL);
+ALTER TABLE nulls_table REPLICA IDENTITY FULL;
+
+CREATE TABLE utf8_table (f1 TEXT PRIMARY KEY, f2 TEXT);
+INSERT INTO utf8_table VALUES ('това е текст', 'това ''е'' "текст"');
+ALTER TABLE utf8_table REPLICA IDENTITY FULL;
+
+CREATE TABLE no_replica_identity (f1 INTEGER);
+INSERT INTO no_replica_identity VALUES (1), (2);
+
+#
+# Error checking
+#
+
+> CREATE MATERIALIZED SOURCE "no_such_host"
+  FROM POSTGRES HOST 'host=no_such_postgres.mtrlz.com port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_host;
+Source error: File IO: error connecting to server: failed to lookup address information: Name or service not known
```

Comment:
A hostname would have been nice here even though the error is long already as it is. I guess this comes from repeately unwrapping Error v.s. throwing an exception.